sdk/storage/azure-storage-blobs/CMakeLists.txt

```diff
@@ -65,7 +65,6 @@ target_include_directories(
     PUBLIC
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
       $<INSTALL_INTERFACE:include>
-      ${azure-storage-common-cpp_INCLUDE_DIRS}
```

Comment:
Hi @antkmsft , storage team would like to understand why this include directory is necessary.

Reply:
It plays a role when you only build blobs library (load `sdk/storage/azure-storage-blobs/CMakeLists.txt` and NOT the CmakeLists in the root folder), and get storage-common as find_package(), i.e. when you build a vcpackage. When you don't build a single library, but the whole SDK, `AZ_ALL_LIBRARIES` is defined, so we don't try to find_package(azure-storage-common-cpp), and build as usual, plus then `azure-storage-common-cpp_INCLUDE_DIRS` is not defined, having no effect.

Reply:
Do not remove this line, it will break vcpkg build. It is an empty string when not defined, should be harmless for you either way.

Reply:
Shouldn't the include dirs of `azure-storage-common` be set already via `target_link_libraries(azure-storage-blobs PUBLIC Azure::azure-storage-common)`?

Reply:
That only adds link instructions, but does not configure include directory. It works when building the entire SDK, because in that case all the include directories are accessible. After all, why then do we have to do target_include_directories(${LIBXML2_INCLUDE_DIRS})? When building as vcpkg, azure-storage-common to azure-storage-blobs is no different than libxml2 or any other 3rd party lib.

Reply:
> That only adds link instructions, but does not configure include directory.

I don't think so. It says in [this page](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#transitive-usage-requirements) `The usage requirements of a target can transitively propagate to dependents.`

[This section](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-specification-and-usage-requirements) and [this section](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#target-properties) have a better definition for `usage requirements`.

Reply:
@antkmsft ping...

This PR is blocked here. Can you take a look? Thanks!